datadog_checks_dev/tests/test_docker.py

```diff
@@ -80,5 +88,8 @@ def test_retry_condition_failed_only_on_first_run(self):
         condition = mock.MagicMock()
         condition.side_effect = [Exception("exception"), None, None]
 
-        with docker_run(compose_file, attempts=3, conditions=[condition]):
-            assert condition.call_count == 2
+        with patch('datadog_checks.dev.docker.ComposeFileUp', return_value=ComposeFileMock()), patch(
```

Comment:
Why do we need `ComposeFileMock()` here? Considering `ComposeFileMock` is essentially a no-op, can't just we omit the `return_value` for these?

Reply:
We serialize the value returned by the up function, using a bare mock make it fail because the mock is not json serializable. I worked around it using 

```
        up = MagicMock()
        up.return_value = ""
```

and dropping the class I had

Reply:
Gotcha 👍. It should also be possible to do `up=mock.MagicMock(return_value="")` on the `docker_run` call for the same effect, right?

Reply:
Yep we can also inline it